src/tensorinfo.cpp

```diff
@@ -30,42 +30,38 @@ namespace ovms {
 TensorInfo::TensorInfo(const std::string& name,
     const Precision& precision,
     const Shape& shape) :
-    name(name),
-    mapping(""),
-    precision(precision),
-    shape(shape),
-    layout(Layout::getDefaultLayout()) {}
+    TensorInfo(name, "", precision, shape, Layout::getDefaultLayout()) {}
 
 TensorInfo::TensorInfo(const std::string& name,
     const Precision& precision,
     const shape_t& shape) :
-    name(name),
-    mapping(""),
-    precision(precision),
-    shape(shape),
-    layout(Layout::getDefaultLayout()) {
-}
+    TensorInfo(name, "", precision, shape, Layout::getDefaultLayout()) {}
 
 TensorInfo::TensorInfo(const std::string& name,
     const ovms::Precision& precision,
     const shape_t& shape,
     const Layout& layout) :
-    name(name),
-    mapping(""),
-    precision(precision),
-    shape(shape),
-    layout(layout) {
-}
+    TensorInfo(name, "", precision, shape, layout) {}
+
+TensorInfo::TensorInfo(const std::string& name,
+    const std::string& mapping,
+    const Precision& precision,
+    const shape_t& shape) :
+    TensorInfo(name, mapping, precision, shape, Layout::getDefaultLayout()) {}
 
 TensorInfo::TensorInfo(const std::string& name,
     const ovms::Precision& precision,
     const Shape& shape,
     const Layout& layout) :
-    name(name),
-    mapping(""),
-    precision(precision),
-    shape(shape),
-    layout(layout) {
+    TensorInfo(name, "", precision, shape, layout) {}
+
+template <class T>
+static BytesType getBytesTypeFromShape(std::vector<T> shape) {
```

Comment:
Each `getBytesTypeFromShape` call creates copy of `shape`. Use `const std::vector<T>& shape` to avoid that